scripts/status_reporter/bnumbers.py

```diff
@@ -0,0 +1,54 @@
+import re
+
+import boto3
+
+
+class BibNumberGenerator:
+    def __init__(
+        self,
+        s3_client,
+        mets_bucket_name="wellcomecollection-assets-workingstorage",
+        mets_only_root_prefix="mets_only/",
+    ):
+        self.s3_client = s3_client
+        self.bucket = mets_bucket_name
+        self.prefix = mets_only_root_prefix
+
+    def _get_matching_s3_objects(self, client, bucket, prefix="", suffix=""):
+        kwargs = {"Bucket": bucket}
+
+        if isinstance(prefix, str):
+            kwargs["Prefix"] = prefix
+
+        while True:
+            resp = client.list_objects_v2(**kwargs)
+
+            try:
+                contents = resp["Contents"]
+            except KeyError:
+                return
+
+            for obj in contents:
+                key = obj["Key"]
+                if key.startswith(prefix) and key.endswith(suffix):
+                    yield obj
+
+            try:
+                kwargs["ContinuationToken"] = resp["NextContinuationToken"]
+            except KeyError:
+                break
```

Comment:
This is oooolllllldddd code, use paginates instead: https://alexwlchan.net/2019/07/listing-s3-keys/

(Also the suffix thing isn’t something we need, we can ditch it)